cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -525,9 +525,8 @@ class GroupByNode : public ExecNode {
   void StopProducing(ExecNode* output) override {
     DCHECK_EQ(output, outputs_[0]);
 
-    if (input_counter_.Cancel()) {
-      finished_.MarkFinished();
-    } else if (output_counter_.Cancel()) {
+    ARROW_UNUSED(input_counter_.Cancel());
```

Comment:
Looks good.  I don't know if you even need to call `input_counter_.Cancel()` here but I also suppose it can't hurt and maybe it will save some extra work.